server/src/test/java/org/elasticsearch/index/replication/RecoveryDuringReplicationTests.java

```diff
@@ -363,24 +367,69 @@ public void testResyncAfterPrimaryPromotion() throws Exception {
 
             final IndexShard oldPrimary = shards.getPrimary();
             final IndexShard newPrimary = shards.getReplicas().get(0);
+            final IndexShard justReplica = shards.getReplicas().get(1);
 
             // simulate docs that were inflight when primary failed
-            final int extraDocs = randomIntBetween(0, 5);
+            final int extraDocs = randomIntBetween(1, 5);
```

Comment:
why not have 0 here?

Reply:
Also, can you double check that 5 is above a single batch of the resync method? I think we want to test multiple iterations.